_examples/jwt_token/jwt/token_verifier_jwt.go

```diff
@@ -229,33 +258,44 @@ func (verifier *TokenVerifier) VerifySubscribeToken(t string) (SubscribeToken, e
 		return SubscribeToken{}, err
 	}
 
+	token, err = jwt.Parse([]byte(t), verifier.selectVerifier(token.Header().Algorithm))
+	if err != nil {
+		return SubscribeToken{}, fmt.Errorf("error verifying subscribe token: %w", err)
+	}
+
 	claims := &subscribeTokenClaims{}
-	err = json.Unmarshal(token.RawClaims(), claims)
+	err = json.Unmarshal(token.Claims(), claims)
 	if err != nil {
 		return SubscribeToken{}, err
 	}
 
 	now := time.Now()
-	if !claims.IsValidExpiresAt(now) || !claims.IsValidNotBefore(now) {
+	if !claims.IsValidExpiresAt(now) {
 		return SubscribeToken{}, ErrTokenExpired
 	}
+	if !claims.IsValidNotBefore(now) {
+		return SubscribeToken{}, errors.New("token not valid yet")
```

Comment:
If I remember right this is important to return ErrTokenExpired in such cases too to trigger automatic token refresh.

Reply:
Though probably returning custom error here is even better - subscription will automatically re-establish without refresh until token become valid. Need to check the exact behavior.

Reply:
This results into disconnect with `invalid token`. Anyway, I suppose it's fine for the example.